### PR TITLE
Minimum array mutation, Repo beautify and Automations

### PR DESCRIPTION
**Pull request by Criticalcarpet**
## Overview
Applied some change in source code and update of README with some extra test and self automating files

**Provide a general overview of this PR:**
Change in main source code according the current js style and use of modern array manipulations without mutations and addition of self automation files such as a batch script followed by a Npm and GPR publishing workflow (which requires to set your own npm token in the repo secrets, for it to work as intended :>)

## General Information
1. **Does it solve a bug?**
No

2. **What changes have you made?**
- Updated files under `src/`
- Updated `package.json`
   - Added `discord.js^13.4.0` for testing
   - Added `dotenv^10.0.0` for testing
- Updated `.eslintrc.json`
- Updated README.md
- Added `test/`
- Added `run.bat`
- Added `.github/workflows/publish.yml`
- Added `src/util/error.js`

3. **What alternatives have you tried?**
- Implemented `.map` instead of initializing a empty array and pushing into it
- Implemented `.filter` instead of `for of` loop exclusions
- Implemented conditional chaining for `undefined` parameters
- Implemented custom errors for `missing parameters` and `send message prohibition`
- Eliminated unnecessary curly braces (`{}`) in `if` loops

### CODE REVIEW

**Comment by RaZeSloth** [comment]
Why are half of the commits a bug emoji?

**Comment by Criticalcarpet** [comment]
That was for my reference, I didn't know any other emote to apply nor a keyword.. but the last commit is meaningful.
sorry for the inconvenience.
BTW is the change meaningful and useful ?

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          npm config list
+          npm publish --access public 
+        env:
+          CI: true
+          NODE_AUTH_TOKEN: ${{secrets.npm_token}}
+
+      # Publish to GitHub Package Registry
+      - uses: actions/setup-node@master
+        with:
+          node-version: 12
```
Comment (RaZeSloth) [inline .github/workflows/publish.yml:33]: iirc discord.js requires v16 and over nodejs version to build.

Comment (Criticalcarpet) [reply .github/workflows/publish.yml:33]: That is only to publish the package.. also discord is in dev dependency...

**Review by ThatsLiamS — CHANGES_REQUESTED** [review]
Remove all added trailing spaces to end of lines as these can have unintended consequences and take up space.

**Review by ThatsLiamS — CHANGES_REQUESTED** [review]
#### Testing directory 

All contributors have different methods to test and as a rule of thumb we recommend testing through `npm link`. Please remove the **[test directory](https://github.com/Criticalcarpet/discord.js-ghost-ping/tree/main/test)** and related dev dependencies.

**Review by ThatsLiamS — CHANGES_REQUESTED** [review]
#### Automation 

While automation is cool _(for the most part)_, this belong in a separate pull request as the changes are unrelated to the primary feature change. 

- **[run.bat](https://github.com/Criticalcarpet/discord.js-ghost-ping/blob/main/run.bat)**
- **[NPM Publish](https://github.com/Criticalcarpet/discord.js-ghost-ping/blob/main/.github/workflows/publish.yml)**

Along with the scripts added in **[package.json](https://github.com/Criticalcarpet/discord.js-ghost-ping/blob/main/package.json)**

**Comment by RaZeSloth** [comment]
I think that all test and publishing scripts could be done with [husky](https://typicode.github.io/husky/#/)

**Comment by Criticalcarpet** [comment]
Yeah... sure

I will create separate PRs' for separate/unique change made... 
example : https://github.com/ThatsLiamS/discord.js-ghost-ping/pull/13

**Pull request closed without merging**